File: app/src/main/java/ru/systempla/weatherapp/di/module/CiceroneModule.java
package ru.systempla.weatherapp.di.module;

import javax.inject.Singleton;

import dagger.Module;
import dagger.Provides;
import ru.terrakok.cicerone.Cicerone;
import ru.terrakok.cicerone.NavigatorHolder;
import ru.terrakok.cicerone.Router;

@SuppressWarnings("WeakerAccess")
@Module
public class CiceroneModule {

    private Cicerone<Router> cicerone = Cicerone.create();

    @Singleton
    @Provides
    public NavigatorHolder navigatorHolder(){
        return cicerone.getNavigatorHolder();
    }

    @Singleton
    @Provides
    public Router router(){
        return cicerone.getRouter();
    }
}

File: app/src/main/java/ru/systempla/weatherapp/di/module/CiceroneModule.kt
package ru.systempla.weatherapp.di.module

import dagger.Module
import dagger.Provides
import ru.terrakok.cicerone.Cicerone
import ru.terrakok.cicerone.NavigatorHolder
import ru.terrakok.cicerone.Router
import javax.inject.Singleton

@Module
class CiceroneModule {
    private val cicerone = Cicerone.create()

    @Singleton
    @Provides
    fun navigatorHolder(): NavigatorHolder = cicerone.navigatorHolder

    @Singleton
    @Provides
    fun router(): Router = cicerone.router
}